### PR TITLE
Add `inAppSearch` for advanced search providers

### PR DESCRIPTION
**Pull request by Altahrim**
Add information about in-app search like `mail` or `talk` applications

### CODE REVIEW

**Review by provokateurin — APPROVED** [review]
OpenAPI is ok

**Comment by ChristophWurst** [comment]
Non-breaking alternative: add a tagging interface with no methods.

Added benefit is that the majority of search providers don't have to implement this new method just to return false :)

**Review by ChristophWurst — COMMENTED** [review]
Commit the new file ;)

Then run `composer dump-autoload` and commit the changes that add the new class to the class map.

File: lib/public/Search/IInAppSearch.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ */
+namespace OCP\Search;
+
+/**
+ * Interface for search providers supporting in-app search
+ *
+ * @since 28.0.0
+ */
+interface IInAppSearch {
```
Comment (ChristophWurst) [inline lib/public/Search/IInAppSearch.php:33]:
```suggestion
interface IInAppSearch extends IProvider {
```

I *think* that the is-a relation applies here. IInAppSearch should only be used by classes that are a search provider.

But I'm okay without. Just dumping my thoughts :)

**Review by ChristophWurst — APPROVED** [review]

File: lib/public/Search/IInAppSearch.php (hunk 1 above)
Comment (Altahrim) [reply lib/public/Search/IInAppSearch.php:33]: I am agree. It makes no sense without it :)

**Comment by Altahrim** [comment]
Unrelated Cypress failure

**Pull request merged**